### PR TITLE
Fix [TeamCity] build

### LINKED ISSUE #3244: TeamCity Build Badge "Invalid response data"
Are you experiencing an issue with...

- [x] [shields.io](https://shields.io/#/)
- [ ] My own instance
- [ ] [gh-badges NPM package](https://www.npmjs.com/package/gh-badges)

:beetle: **Description**

I tried to setup a badge for my TeamCity build status.
I'm pretty sure I did everything right.
However, the error is "Invalid Response Data".

:link: **Link to the badge**

https://img.shields.io/teamcity/https/my.teamcity.url/e/My_Build.svg?style=flat-square
I don't want to publish my TeamCity instance, because it is private.

:bulb: **Possible Solution**

I compared the API endpoints of https://teamcity.jetbrains.com and my server and noticed that the `guest` parameter must be set to 1 to get access. Otherwise, you get a 403 response.
It would work if you just add `guest=1` [here](https://github.com/badges/shields/blob/master/services/teamcity/teamcity-base.js#L20).
Solution

I could PR it if you want.

### PR DESCRIPTION
**Pull request by calebcartwright**
Fix #3244 

Updated schema validation on `statusText` field to account for additional text suffixes

### CODE REVIEW

**Comment by calebcartwright** [comment]
@Stupremee can you help validate this update by checking your TeamCity build in the staging app @ https://shields-staging-pr-3245.herokuapp.com/

**Comment by Stupremee** [comment]
Works fine. 👍

**Comment by calebcartwright** [comment]
Perfect, thanks @Stupremee!

Once this PR is approved by a maintainer, it will be merged and deployed. The deployment status will be reflected in this PR (once merged) so you can track this PR to find out once the fix has been deployed

File: services/teamcity/teamcity-build.service.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 const Joi = require('joi')
 const TeamCityBase = require('./teamcity-base')
 
-const buildStatusTextRegex = /^Success|Failure|Error|Tests( failed: \d+( \(\d+ new\))?)?(,)?( passed: \d+)?(,)?( ignored: \d+)?(,)?( muted: \d+)?$/
+const buildStatusTextRegex = /^Success|Failure|Error|Tests( failed: \d+( \(\d+ new\))?)?(,)?( passed: \d+)?(,)?( ignored: \d+)?(,)?( muted: \d+)?(.*)?$/
```
Comment (paulmelnikow) [inline services/teamcity/teamcity-build.service.js]: I think removing the `$` should be equivalent:

```suggestion
const buildStatusTextRegex = /^Success|Failure|Error|Tests( failed: \d+( \(\d+ new\))?)?(,)?( passed: \d+)?(,)?( ignored: \d+)?(,)?( muted: \d+)?/
```

Also could you add a note about the issue this is solving (and a link)?

Comment (Stupremee) [reply services/teamcity/teamcity-build.service.js]: https://github.com/badges/shields/issues/3244 
here you go

Comment (paulmelnikow) [reply services/teamcity/teamcity-build.service.js]: Ah, thanks! I meant that it'd be good for @calebcartwright to include an explanation in the code of why the isn't matching the entire string.

**Pull request merged**